### PR TITLE
Refactors dog clothing

### LINKED ISSUE #17068: Ian clothing is getting out of hand (codewise)
We need to replace the HUGE switch()s with vars like ```dog_name```,```dog_desc```,```dog_phrases```,```dog_icon``` and ```dog_icon_state``` on the ```/obj/item/clothing``` type

@coiax has volunteered to do this.

### PR DESCRIPTION
**Pull request by coiax**
Fixes #17068.

:cl: coiax
rscadd: Clown mask appearance toggle now has an action button.
/:cl:

### CODE REVIEW

**Comment by RemieRichards** [comment]
I also said ```dog_icon``` and ```dog_icon_state``` so that Ian's clothing files don't *HAVE* to be humongous and take 70 years to load.

**Comment by Incoming5643** [comment]
A refactor is the last thing that needs a changelog. Player wouln't (shouldn't) even notice this.

**Comment by coiax** [comment]
I think I've heard it discussed that making some visibility of coder's work on UNDER THE HOOD stuff means people will be more aware of where our effort's being spent. Also, this will probably introduce some bugs, probably where you'd be able to put a hat on Ian and have the wrong sprite.

**Comment by coiax** [comment]
Also I just pushed my system which uses the new /datum/dog_fashion to store all this stuff

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -248,10 +248,11 @@
 	// First, change back to defaults
 	name = real_name
 	desc = initial(desc)
-	speak = initial(speak)
-	speak_emote = initial(speak_emote)
-	emote_hear = initial(emote_hear)
-	emote_see = initial(emote_see)
+	// BYOND/DM doesn't support the use of initial on lists.
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: dont write this, we know.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 2]
```diff
@@ -267,143 +245,24 @@
...
-				name = "[real_name] the Clown"
-				desc = "Honkman's best friend."
-				speak = list("HONK!", "Honk!")
-				emote_see = list("plays tricks.", "slips.")
-			else
-				special_hat = 0
+	// First, change back to defaults
+	name = real_name
+	desc = initial(desc)
+	// BYOND/DM doesn't support the use of initial on lists.
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: dont write this, we know.

Comment (coiax) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: doesn't specify in the DM reference

**Comment by Incoming5643** [comment]
#13382

**Comment by coiax** [comment]
Fine.

File: code/game/objects/items/devices/radio/beacon.dm [hunk 3]
```diff
@@ -5,6 +5,7 @@
 	item_state = "beacon"
 	var/code = "electronic"
 	origin_tech = "bluespace=1"
+	dog_fashion = null
```
Comment (optimumtact) [inline code/game/objects/items/devices/radio/beacon.dm]: What is up with these overrides that are just also null?

**Comment by optimumtact** [comment]
Ps this is a great example of component based design and we should make everyone read it

File: code/game/objects/items/devices/radio/beacon.dm (hunk 3 above)
Comment (coiax) [inline code/game/objects/items/devices/radio/beacon.dm]: Dogs can wear station radios, but they can't wear any of their children objects. In this particular case, inheritance is a curse, but short of manually specifying a list of types to allow (which ruins the whole point), will just have to override children with no sprites manually. The alternative is allow children, but that will mean the sprite is wrong.

File: code/datums/dog_fashion.dm [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	name = "Runtime"
+	emote_see = list("coughs up a furball", "stretches")
+	emote_hear = list("purrs")
+	speak = list("Purrr", "Meow!", "MAOOOOOW!", "HISSSSS", "MEEEEEEW")
+	desc = "It's a cute little kitty-cat! ... wait ... what the hell?"
+
+/datum/dog_fashion/head/rabbit
+	name = "Hoppy"
+	emote_see = list("twitches its nose", "hops around a bit")
+	desc = "This is Hoppy. It's a corgi-...urmm... bunny rabbit"
```
Comment (ChangelingRain) [inline code/datums/dog_fashion.dm]: Add a `.` at the end.

File: code/datums/dog_fashion.dm [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+	emote_hear = list("lights the way!", "illuminates.", "yaps!")
+	desc = "He has a very shiny nose."
+
+/datum/dog_fashion/head/sombrero
+	name = "Segnor REAL_NAME"
+	desc = "You must respect Elder Dogname"
+
+/datum/dog_fashion/head/sombrero/New(mob/M)
+	..()
+	desc = "You must respect Elder [M.real_name]"
```
Comment (ChangelingRain) [inline code/datums/dog_fashion.dm]: Add a `.` at the end.

File: code/__DEFINES/clothing.dm [hunk 6]
```diff
@@ -101,3 +101,4 @@
 
 #define TINT_DARKENED 2			//Threshold of tint level to apply weld mask overlay
 #define TINT_BLIND 3			//Threshold of tint level to obscure vision fully
+
```
Comment (ChangelingRain) [inline code/__DEFINES/clothing.dm]: NO RANDOM LINES

**Comment by coiax** [comment]
I have been thinking about this of recent, and I'm kinda unhappy how I added a var on all /obj/item. But I'm unsure how to refactor it without resorting to a giant if statement with a bunch of typechecks.

**Pull request merged**